File: seedsanskriti_backend/src/main/java/com/seedsanskriti/service/CartService.java
package com.seedsanskriti.service;

import org.springframework.security.core.Authentication;

import com.seedsanskriti.dto.AddToCartRequest;
import com.seedsanskriti.dto.CartResponse;

import com.seedsanskriti.dto.UpdateCartRequest;

public interface CartService {

	String addToCart(AddToCartRequest request,
            Authentication authentication);
	
	CartResponse getMyCart(Authentication authentication);
	
	String updateCartItem(UpdateCartRequest request,Authentication authentication);
	
	String removeCartItem(Long cartItemId,
            Authentication authentication);
	
	String clearCart(Authentication authentication);
	
	
	
}

File: seedsanskriti_frontend/src/services/cartService.js
import apiClient from './apiClient';

const cartService = {
  getMyCart: () => apiClient.get('/cart/my-cart').then((res) => res.data),

  addToCart: (payload) => apiClient.post('/cart/add', payload).then((res) => res.data),

  updateCartItem: (payload) => apiClient.put('/cart/update', payload).then((res) => res.data),

  removeCartItem: (cartItemId) =>
    apiClient.delete(`/cart/remove/${cartItemId}`).then((res) => res.data),

  clearCart: () => apiClient.delete('/cart/clear').then((res) => res.data),
};

export default cartService;
